src/visor/readable.go

```diff
@@ -443,7 +454,7 @@ func NewGenesisReadableTransaction(t *Transaction) (*ReadableTransaction, error)
 }
 
 // NewReadableTransaction creates readable transaction
-func NewReadableTransaction(t *Transaction) (*ReadableTransaction, error) {
+func NewReadableTransaction(t *Transaction, inData []*historydb.UxOut) (*ReadableTransaction, error) {
```

Comment:
By adding the `inData` parameter to `NewReadableTransaction`, now all the functions that want to obtain a transaction in the appropriate format for a JSON response must send an slice with the details of the inputs. This requirement was put in place to mantain consistency in the api and was the reason why all the endpoints that return transactions were modified.

The situation of the `NewReadableBlock` and `NewReadableBlocks` is similar.

An important detail is that this function could receive a gateway as a parameter and thus autonomously search the inputs data, without having to always do it before calling `NewReadableTransaction`, which would allow to eliminate repeated code. It was not done that way to maintain the format, since in this file there are no functions that autonomously search for data. However, if desired, the modification can be made.